File: scripts/list_courses.py
"""
Gitea List Courses
Lists all courses (organizations) visible to the authenticated user.

Usage:
  python list_courses.py
"""

from config import GITEA_URL, get_credentials
from gitea_api import get_session


def main():
    admin_user, admin_pass = get_credentials()
    session = get_session(admin_user, admin_pass)

    courses = []
    page = 1
    while True:
        resp = session.get(
            f"{GITEA_URL}/api/v1/admin/orgs",
            params={"page": page, "limit": 50},
        )
        if resp.status_code != 200:
            resp = session.get(
                f"{GITEA_URL}/api/v1/user/orgs",
                params={"page": page, "limit": 50},
            )
            if resp.status_code != 200:
                print(f"Error: Could not list courses (HTTP {resp.status_code})")
                break
        data = resp.json()
        if not data:
            break
        courses.extend(data)
        page += 1

    if not courses:
        print("No courses found.")
        return

    print(f"Courses ({len(courses)}):\n")
    for course in courses:
        vis = course.get("visibility", "unknown")
        desc = course.get("description") or ""
        print(f"  {course['username']} ({vis}){('  ' + desc) if desc else ''}")


if __name__ == "__main__":
    main()

File: scripts/list_courses.ts
/**
 * Gitea List Courses
 * Lists all courses (organizations) visible to the authenticated user.
 *
 * Usage: node list_courses.js
 */

import { GITEA_URL, get_credentials } from './config.ts';
import { get_session } from './gitea_api.ts';

async function main(): Promise<void> {
  const [admin_user, admin_pass] = await get_credentials();
  const session = get_session(admin_user, admin_pass);

  const courses = [];
  let page = 1;
  while (true) {
    let resp = await session.get(`${GITEA_URL}/api/v1/admin/orgs`, { page, limit: 50 });
    if (resp.status !== 200) {
      resp = await session.get(`${GITEA_URL}/api/v1/user/orgs`, { page, limit: 50 });
      if (resp.status !== 200) {
        console.log(`Error: Could not list courses (HTTP ${resp.status})`);
        break;
      }
    }
    const data = await resp.json();
    if (!data.length) break;
    courses.push(...data);
    page++;
  }

  if (!courses.length) {
    console.log('No courses found.');
    return;
  }

  console.log(`Courses (${courses.length}):\n`);
  for (const course of courses) {
    const vis = course.visibility || 'unknown';
    const desc = course.description || '';
    console.log(`  ${course.username} (${vis})${desc ? '  ' + desc : ''}`);
  }
}

main();
